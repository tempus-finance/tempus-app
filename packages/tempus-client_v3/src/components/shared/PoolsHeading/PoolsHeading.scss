.tc__poolsHeading {
  position: relative;
  width: max-content;
  height: 40px;
  padding-right: 8px;
}

.tc__poolsHeading-content {
  height: 32px;
  display: flex;
  align-items: center;

  .tc__btn {
    margin-right: 8px;
  }
}

.tc__poolsHeading-overlay {
  position: absolute;
  height: 24px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  background-color: var(--poolsHeadingOverlay);
}
