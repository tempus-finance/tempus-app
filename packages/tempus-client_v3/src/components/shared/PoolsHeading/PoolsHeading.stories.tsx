import { ComponentStory, ComponentMeta } from '@storybook/react';

import PoolsHeading from './PoolsHeading';

export default {
  title: 'PoolsHeading',
  component: PoolsHeading,
  argTypes: {
    text: {
      control: {
        type: 'text',
      },
    },
    onBackButtonClick: {
      required: false,
      defaultValue: undefined,
    },
  },
} as ComponentMeta<typeof PoolsHeading>;

const style = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  height: '100px',
};

const Template: ComponentStory<typeof PoolsHeading> = args => (
  <div style={style}>
    <PoolsHeading {...args} />
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  text: 'Ethereum-network pools',
};
