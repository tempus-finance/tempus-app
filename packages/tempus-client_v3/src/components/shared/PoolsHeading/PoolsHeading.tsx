import { FC } from 'react';
import Typography from '../Typography';

import './PoolsHeading.scss';

interface PoolsHeadingProps {
  text: string;
}

const PoolsHeading: FC<PoolsHeadingProps> = (props): JSX.Element => {
  const { text } = props;

  return (
    <div className="tc__poolsHeading">
      <div className="tc__poolsHeading-content">
        <Typography variant="title" weight="bold">
          {text}
        </Typography>
      </div>
      <div className="tc__poolsHeading-overlay" />
    </div>
  );
};
export default PoolsHeading;
