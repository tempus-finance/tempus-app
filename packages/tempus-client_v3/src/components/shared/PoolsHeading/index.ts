import PoolsHeading from './PoolsHeading';

export default PoolsHeading;
