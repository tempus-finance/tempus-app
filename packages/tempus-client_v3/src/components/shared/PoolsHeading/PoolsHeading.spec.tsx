import { render } from '@testing-library/react';
import PoolsHeading, { PoolsHeadingProps } from './PoolsHeading';

const defaultProps: PoolsHeadingProps = {
  text: 'Ethereum-network pools',
};

const subject = (props: PoolsHeadingProps) => render(<PoolsHeading {...props} />);

describe('PoolsHeading', () => {
  it('renders a heading label with provided text', () => {
    const { container, getByText } = subject(defaultProps);

    const actualText = getByText('Ethereum-network pools');

    expect(actualText).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
