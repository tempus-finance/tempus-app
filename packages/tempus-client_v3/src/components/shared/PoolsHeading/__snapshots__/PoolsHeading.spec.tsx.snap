// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoolsHeading renders a heading label with provided text 1`] = `
<div>
  <div
    class="tc__poolsHeading"
  >
    <div
      class="tc__poolsHeading-content"
    >
      <div
        style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Ethereum-network pools
      </div>
    </div>
    <div
      class="tc__poolsHeading-overlay"
    />
  </div>
</div>
`;
